### PR TITLE
Fix #4077

### LINKED ISSUE #4077: V4.2.0 introduced issue in cypher generation
**Describe the bug**
Don't exactly know what's happening, but it seems to be related with authorization.

**Type definitions**
```graphql
type PreviewClip @mutation(operations: [DELETE]) {
    id: ID! @id
    startTime: Int!
    duration: Int!

    markedAsDone: Boolean! @default(value: false)

    clippedFrom: Video! @relationship(type: "VIDEO_HAS_PREVIEW_CLIP", direction: IN)

    creationDate: DateTime! @timestamp(operations: [CREATE])
    lastUpdate: DateTime! @timestamp(operations: [CREATE, UPDATE])
}
# Rules
extend type PreviewClip
    @authorization(
        filter: [
            { where: { node: { clippedFrom: { publisher: { id: "$jwt.sub" } } } } }
            { where: { jwt: { roles_INCLUDES: "admin" } } }
        ]
    )

type Video @mutation(operations: [UPDATE]) {
    id: ID! @id

    publisher: User! @relationship(type: "PUBLISHER", direction: IN)

    creationDate: DateTime! @timestamp(operations: [CREATE])
    lastUpdate: DateTime! @timestamp(operations: [CREATE, UPDATE])
}
# Rules
extend type Video
    @authorization(
        filter: [
            { where: { node: { publisher: { id: "$jwt.sub" } } } }
            { where: { jwt: { roles_INCLUDES: "admin" } } }
            {
                requireAuthentication: false
                operations: [READ]
                where: { node: { processing: "published" } }
            }
        ]
    )
    
type User @mutation(operations: [UPDATE]) {
    id: ID! @id
}

# Rules
extend type User
    @authorization(
        validate: [
            { operations: [UPDATE], where: { node: { id: "$jwt.sub" } } }
            { operations: [UPDATE], where: { jwt: { roles_INCLUDES: "admin" } } }
        ]
    )
```

**To Reproduce**

1. Execute the following Query
    ```graphql
    query listPossiblePreviewClips($videoId: ID!) {
      previewClips(where: { clippedFrom: { id: $videoId }, NOT: { markedAsDone: true } }) {
        id
      }
    }
    ```
2. See error
```
{
  message: 'Variable `clippedFromCount` not defined (line 11, column 9 (offset: 405))\n' +
    '"WHERE ((clippedFromCount <> 0 AND this0.id = $param1) AND (($isAuthenticated = true AND var3 = true) OR ($isAuthenticated = true AND $param3 IN $jwt.roles)))"\n' +
    '         ^',
  locations: [ { line: 2, column: 3 } ],
  path: [ 'previewClips' ],
  extensions: {
    code: 'INTERNAL_SERVER_ERROR',
    stacktrace: [
      'Neo4jError: Variable `clippedFromCount` not defined (line 11, column 9 (offset: 405))',
      '"WHERE ((clippedFromCount <> 0 AND this0.id = $param1) AND (($isAuthenticated = true AND var3 = true) OR ($isAuthenticated = true AND $param3 IN $jwt.roles)))"',
      '         ^',
      '',
      '    at captureStacktrace (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/result.js:613:17)',
      '    at new Result (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/result.js:105:23)',
      '    at newCompletedResult (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:510:12)',
      '    at Object.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:342:20)',
      '    at Transaction.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction.js:183:34)',
      '    at ManagedTransaction.run (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/transaction-managed.js:56:21)',
      '    at Executor.transactionRun (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/dist/classes/Executor.js:162:28)',
      '    at /home/mathix/projects/peach_api/node_modules/@neo4j/graphql/dist/classes/Executor.js:143:33',
      '    at TransactionExecutor._safeExecuteTransactionWork (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:213:26)',
      '    at TransactionExecutor.<anonymous> (/home/mathix/projects/peach_api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:200:46)'
    ]
  }
}
```

**Expected behavior**
Work as expected in 4.1.3

**System (please complete the following information):**

* OS: Archlinux
* Version: @neo4j/graphql@4.2.0
* Node.js version: v18.16.1

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Fix auth filters on new translation layer

## Complexity

Complexity: Low

# Issue

Closes #4077

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4084.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 1]
```diff
@@ -132,8 +134,8 @@ export class ReadOperation extends Operation {
 
         const clause = Cypher.concat(
             ...preSelection,
-            matchClause,
             ...authFilterSubqueries,
+            matchClause,
```
Comment (darrellwarde) [inline packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts:138]: Might the `authFilterSubqueries` not require a variable from the `matchClause` out of interest?

**Review by darrellwarde — APPROVED** [review]
LGTM!

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts:138]: In theory not, as these should be in the preSelection

**Pull request merged**